conans/client/toolchain/msbuild.py

```diff
@@ -0,0 +1,99 @@
+import os
+import textwrap
+from xml.dom import minidom
+
+from conans.client.tools import msvs_toolset
+from conans.util.files import save, load
+
+
+class MSBuildToolchain(object):
+
+    def __init__(self, conanfile):
+        self._conanfile = conanfile
+        self.definitions = {}
+
+    @ staticmethod
+    def _name_condition(settings):
+        toolset = msvs_toolset(settings)
+
+        props = [("Configuration", settings.build_type),
+                 # FIXME: This probably requires mapping ARM architectures
+                 ("Platform", {'x86': 'Win32',
+                               'x86_64': 'x64'}.get(settings.get_safe("arch"))),
+                 ("PlatformToolset", toolset)]
+
+        name = "".join("_%s" % v for _, v in props if v is not None)
+        condition = " And ".join("'$(%s)' == '%s'" % (k, v) for k, v in props if v is not None)
+        return name.lower(), condition
+
+    def write_toolchain_files(self):
+        name, condition = self._name_condition(self._conanfile.settings)
+        config_filename = "conan_toolchain{}.props".format(name)
+        self._write_config_toolchain(config_filename)
```

Comment:
I didn't pay attention to it previously, but now I have a question about toolchains in general (not just msbuild).
we always just override file from scratch, with no warning, if it exists and has local modifications.
is it intended?

Reply:
Yes, the toolchain files should be considered temporary files, probably not to be committed to source control. And when you do a ``conan install`` they can get overwritten. In this one, as it is a multi-config toolchain, new files can be added with the right conditionals, instead of overwriting the existing ones.

Reply:
And yes, automatically generated code, we could add the typical "DO NOT MODIFY" warning.

Reply:
yes, I think warning makes sense then - as I remember you promoted toolchains feature for local flow usage - where you can open the same project without conan and get the exactly same result as with conan invocation

Reply:
maybe also prefix toolchain files with dot to avoid them accidentally checked into git, etc.

Reply:
> Yes, the toolchain files should be considered temporary files, probably not to be committed to source control. 

I think we should reconsider this position. There are a lot of use cases where people want to commit conan-generated stuff back to SCM (many mono-repo related).  I think one of the attractive things about the new toolchain abstraction is that the variables are not path-related like variables from generators, and thus they might actually be OK to commit to SCM for those cases. 

I definitely like the warning idea.

Reply:
@solvingj  in this case, if we want to allow such a files in SCM, we have to be very careful about them.
e.g. we should ensure they never contain absolute paths, which will basically render in useless files for other developers or CI bots for whom paths might differ. and lots of conflicts (pain) for anyone how wanted to update these files.

Reply:
- Added the DO NOT MODIFY comment note warning
- Not renaming to start with dot. It is ugly for usage/consumption, and files starting with dot are not gitignored by default anyway